src/result.rs

```diff
@@ -70,59 +78,79 @@ pub enum CqlType {
 #[napi]
 impl QueryResultWrapper {
     pub fn from_query(internal: QueryResult) -> QueryResultWrapper {
-        QueryResultWrapper { internal }
+        let value = match internal.clone().into_rows_result() {
+            Ok(v) => QueryResultVariant::RowsResult(v),
+            Err(_) => QueryResultVariant::EmptyResult(internal),
+        };
+        QueryResultWrapper { internal: value }
     }
 
     #[napi]
-    pub fn get_rows(&self) -> Option<Vec<RowWrapper>> {
-        let rows = match &self.internal.rows {
-            Some(r) => r,
-            None => {
-                return None;
+    pub fn get_rows(&self) -> napi::Result<Option<Vec<RowWrapper>>> {
```

Comment:
So you changed this function's signature and didn't update any calling code. Is this function currently unused?

Reply:
It's used from nodeJS